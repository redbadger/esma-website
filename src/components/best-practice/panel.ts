type PanelMember = {
  name: string;
  role: string;
};

export type PanelSection = {
  name: string;
  members: PanelMember[];
};

export type Panel = {
  business: PanelSection;
  publicSector: PanelSection;
  socialMobilityAlliance: PanelSection;
  thirdSector: PanelSection;
};

const business: PanelSection = {
  name: "Business",
  members: [
    {
      name: "Helen Mahy CBE",
      role:
        "Non Executive Director at SSE plc",
    },
    {
      name: "Sir Kenneth Olisa OBE",
      role:
        "Chairman Of The Board at Alitam plc",
    },
    {
      name: "Andrew Garard",
      role:
        "Group General Counsel and Director of Corporate Affairs, Meggitt plc",
    },
    { name: "David Ampaw", role: "Partner, DLA Piper & PRIME Board member" },
    { name: "Dr Tony Munton", role: "Managing Director theRTK" },
    { name: "Hollie Crompton", role: "Social Mobility Operations Lead, PwC" },
    { name: "Jeanie York", role: "CTIO, Virgin Media" },
    { name: "John Godfrey", role: "Commercial Director, Red Badger" },
    { name: "Joe Seddon", role: "Founder and CEO, Zero Gravity Tech" },
    { name: "Katherine Ainely", role: "Managing Director Ventures, BT" },
    {
      name: "Katie Perrior",
      role: "former Downing Street Director of Communications",
    },
    {
      name: "Kirsty Cooper",
      role: `General Counsel and Company Secretary, Aviva Plc & Insurance and
        Pensions Champion, Government Dormant Assets Scheme`,
    },
    { name: "Raphael Mokades", role: "Founder & CEO, Rare Recruitment" },
  ],
};

const publicSector: PanelSection = {
  name: "Public Bodies & Education",
  members: [
    {
      name: "Steve Holliday",
      role:
        "Board Member, Careers and Enterprise Company, Chairman of Zenobe, the President of the Energy Institute and Chairman of the Board of Trustees at Crisis",
    },
    {
      name: "Nigel Boardman",
      role:
        "Non-executive Director at the Department for Business, Energy and Industrial Strategy & Vice President of Save the Children",
    },
    {
      name: "Simon Levine",
      role: "CEO, DLA Piper & Office for Students Board Member",
    },
    {
      name: "Paul Cleal",
      role:
        "Former Social Mobility Commissioner, Vice Chair, Kingston University and NCS Trustee",
    },
    {
      name: "Aimee Higgins",
      role: "Executive Director, Careers and Enterprise Company",
    },
    {
      name: "Clare Hutchinson",
      role: "Executive Director, Careers and Enterprise Company",
    },
    { name: "Laura Rawstorne", role: "Social Mobility Lead, DfE" },
    { name: "Paul Botton", role: "Senior Teacher, Westminster School" },
    {
      name: "David Willis",
      role: "Chair, Paradigm Trust and SRA Board Member",
    },
    {
      name: "Likhon Muhammed",
      role: "Head of College, Headteacher at Bourne End E-ACT Academy",
    },
    { name: "Naim Moukarzel", role: "Director of Programme, NCS" },
    { name: "Dr Trevor Richards", role: "Head of All Hallows School" },
    {
      name: "Kirsty Watt",
      role: "Head of Academy Ambassadors, New Schools Network",
    },
    {
      name: "Michelle Lord",
      role:
        "Senior Arts Development Officer and Social Mobility Lead, Cambridge City Council",
    },
    {
      name: "Professor Richard Moorhead",
      role: "Head of Law, Exeter University",
    },
    {
      name: "Professor Steven Vaughan",
      role: "Law and Professional Ethics, UCL",
    },
    {
      name: "Sue Williams QPM Commander",
      role: "Head of Profession – Safeguarding, Metropolitan Police Service ",
    },
    {
      name: "Professor Thom Brooks",
      role: "Dean & Chair in Law and Government at Durham University",
    },
    {
      name: "Sean Anstee",
      role:
        "Executive Director for Advisory, Public Affairs & PR, Cratus and former Leader of Trafford Council",
    },
  ],
};

const socialMobilityAlliance: PanelSection = {
  name: "Social Mobility Scheme Alliance Representatives",
  members: [
    { name: "Liberty King", role: "Positive Transformation Alumni" },
    {
      name: "Tarnpreet Bassi",
      role: "Future Shoosmiths Trainee and SMBP Volunteer",
    },
    {
      name: "Holly Moore",
      role: "ITV Solicitor Apprentice and former SMBP Cluster Head",
    },
    { name: "Shama Aktar", role: "SMBP Alumni" },
    { name: "Muhammad Gangat", role: "SMBP Alumni" },
    { name: "Symran Gakhal", role: "SMBP Alumni" },
    {
      name: "James Hutchens",
      role:
        "Trainee Solicitor and Youth Advisory Board, Aspiring Solicitors and former SMBP Operations Manager",
    },
    { name: "Claud Williams", role: "Aleto Foundation" },
    { name: "Deborah Igunma", role: "Aleto Foundation" },
    { name: "Janira Borges", role: "Aleto Foundation" },
    { name: "Maame Serwaah", role: "Aleto Foundation" },
    { name: "Obi Okereke", role: "Aleto Foundation" },
  ],
};
const thirdSector: PanelSection = {
  name: "Third Sector",
  members: [
    {
      name: "Martin Lewis OBE",
      role: `SMBP Patron and Founder, <a href="https://www.moneyandmentalhealth.org/">The Money and Mental Health Policy Institute</a>`,
    },
    {
      name: "Nick Newman",
      role: "Founder and CEO, National Careers Week CIC",
    },
    { name: "Sherry Coutu CBE ", role: "Workfinder" },
    {
      name: "Paul Evans",
      role: "CEO, Leadership Through Sport and Business",
    },
    { name: "John Shiels", role: "CEO, Manchester United Foundation" },
    {
      name: "Sharon Evans",
      role: "Trustee, PTI and CEO, Dotcom Children’s Foundation",
    },
    {
      name: "Mark Soden",
      role: "SMBP Patron and Director, Cleartrack Performance",
    },
    { name: "Binda Patel", role: "Head of Innovation, The Sutton Trust" },
    { name: "Nicholas Cheffings", role: "Chair, Making the Leap" },
    {
      name: "Sarah Horner",
      role: "Head of Policy and Communications, Learning and Work Institute",
    },
    { name: "Debbie Forster", role: "CEO, Tech Talent Charter" },
  ],
};

const panel: Panel = {
  business,
  publicSector,
  socialMobilityAlliance,
  thirdSector,
};

export default panel;
