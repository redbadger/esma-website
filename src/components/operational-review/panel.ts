type PanelMember = {
  name: string;
  role: string;
};

export type PanelSection = {
  name: string;
  members: PanelMember[];
};

export type Panel = {
  business: PanelSection;
  publicSector: PanelSection;
  socialMobilityAlliance: PanelSection;
  thirdSector: PanelSection;
};

const business: PanelSection = {
  name: "Business",
  members: [
    {
      name: "Andrew Garard",
      role:
        "Group General Counsel and Director of Corporate Affairs, Meggitt plc",
    },
    { name: "David Ampaw", role: "Partner, DLA Piper & PRIME Board member" },
    { name: "Dr Tony Munton", role: "Managing Director theRTK" },
    { name: "Hollie Crompton", role: "Social Mobility Operations Lead, PwC" },
    { name: "Jeanie York", role: "CTIO, Virgin Media" },
    { name: "John Godfrey", role: "Commercial Director, Red Badger" },
    { name: "Joe Seddon", role: "Founder and CEO, Zero Gravity Tech" },
    { name: "Katherine Ainely", role: "Managing Director Ventures, BT" },
    {
      name: "Katie Perrior",
      role: "former Downing Street Director of Communications",
    },
    {
      name: "Kirsty Cooper",
      role: `General Counsel and Company Secretary, Aviva Plc & Insurance and
        Pensions Champion, Government Dormant Assets Scheme`,
    },
    { name: "Raphael Mokades", role: "Founder & CEO, Rare Recruitment" },
  ],
};

const publicSector: PanelSection = {
  name: "Public Bodies & Education",
  members: [],
};
const socialMobilityAlliance: PanelSection = {
  name: "Social Mobility Scheme Alliance Representatives",
  members: [],
};
const thirdSector: PanelSection = {
  name: "Third Sector",
  members: [
    {
      name: "Martin Lewis OBE",
      role: `SMBP Patron and Founder, <a href="https://www.moneyandmentalhealth.org/">The Money and Mental Health Policy Institute</a>`,
    },
    {
      name: "Nick Newman",
      role: "Founder and CEO, National Careers Week CIC",
    },
    { name: "Sherry Coutu CBE ", role: "Workfinder" },
    {
      name: "Paul Evans",
      role: "CEO, Leadership Through Sport and Business",
    },
    { name: "John Shiels", role: "CEO, Manchester United Foundation" },
    {
      name: "Sharon Evans",
      role: "Trustee, PTI and CEO, Dotcom Childrenâ€™s Foundation",
    },
    {
      name: "Mark Soden",
      role: "SMBP Patron and Director, Cleartrack Performance",
    },
    { name: "Binda Patel", role: "Head of Innovation, The Sutton Trust" },
    { name: "Nicholas Cheffings", role: "Chair, Making the Leap" },
    {
      name: "Sarah Horner",
      role: "Head of Policy and Communications, Learning and Work Institute",
    },
    { name: "Debbie Forster", role: "CEO, Tech Talent Charter" },
  ],
};

const panel: Panel = {
  business,
  publicSector,
  socialMobilityAlliance,
  thirdSector,
};

export default panel;
