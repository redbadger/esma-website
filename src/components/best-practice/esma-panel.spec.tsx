import React from "react";
import renderer from "react-test-renderer";
import EsmaPanel from "./esma-panel";
import { Panel } from "./panel";

const samplePanel: Panel = {
  business: {
    name: "Business People",
    members: [
      {
        name: "Derek Trotter",
        role: "CEO of Trotters Independent Traders Ltd",
      },
    ],
  },
  publicSector: {
    name: "Public Sector People",
    members: [
      {
        name: "Jim Hacker",
        role: "Former minister, Department of Administrative Affairs",
      },
    ],
  },
  socialMobilityAlliance: {
    name: "Social Mobility Alliance People",
    members: [
      {
        name: "Barry Matthews",
        role: "SMBP Alumni",
      },
    ],
  },
  thirdSector: {
    name: "Third Sector People",
    members: [
      {
        name: "Bruce Wayne",
        role: "Founder, Thomas and Martha Wayne Foundation",
      },
    ],
  },
};

describe("EsmaPanel", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<EsmaPanel panel={samplePanel} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
