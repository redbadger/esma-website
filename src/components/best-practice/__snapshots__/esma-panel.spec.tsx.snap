// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsmaPanel renders correctly 1`] = `
<section
  className="css-1c7sk15-panelCss"
>
  <header>
    <h2>
      ESMA Panel
    </h2>
    <p>
      The energy and interest for ESMA has been incredible attracting the support of many senior and inspirational leaders on social mobility.
    </p>
  </header>
  <div
    className="panel-members"
  >
    <div
      className="css-1jk3mwe-switcherCss"
    >
      <div
        className="preview"
        onClick={[Function]}
      >
        <h3>
          Business People
        </h3>
        <span
          className="icon "
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-up fa-w-14 "
            data-icon="chevron-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <ul
        className=""
      >
        <li
          className="active"
          onClick={[Function]}
        >
          Business People
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Public Sector People
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Social Mobility Alliance People
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Third Sector People
        </li>
      </ul>
    </div>
    <ul>
      <li>
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <strong
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 0,
              }
            }
          >
            Derek Trotter
          </strong>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "CEO of Trotters Independent Traders Ltd",
              }
            }
            style={
              Object {
                "flexGrow": 1,
              }
            }
          />
        </div>
      </li>
    </ul>
  </div>
  <p>
    Authored by Dr Catherine McGregor, the ESMA panel will produce a guide which summarises key recommendations identified by the Life Stages teams. ESMA also hopes to raise an incubator fund to enable the sharing of best practice materials, and support creative and radical innovation that has the promise to improve people's lives across the UK.
  </p>
</section>
`;
